### PR TITLE
Aemceds 32 footer

### PR DESCRIPTION
**Pull request by akki761**
Fixes #[AEMCEDS-32](https://jira.corp.adobe.com/browse/AEMCEDS-32)

Test URLs:
- Before: https://main--aem-competency--aem-comp.hlx.live/
- After: https://AEMCEDS-32-footer--aem-competency--aem-comp.hlx.live/

- [ ]  **If you are adding a new block or a variation to an existing block**, please fill below:

  Block library path: https://AEMCEDS-32-footer--aem-competency--aem-comp.hlx.live/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/footer&index=0

### CODE REVIEW

File: styles/styles.css [hunk 1]
```diff
@@ -19,7 +19,8 @@
   --body-text-color: #fff;
   --background-color: #002651;
   --background-color-gradient: #000810;
-  --background-color-footer: var(--text-color);
+  --background-color-footer: #000A14;
```
Comment (mkbansal1) [inline styles/styles.css]: I have checked with Designer, footer bg color will be black

File: blocks/footer/footer.js [hunk 2]
```diff
@@ -1,20 +1,62 @@
...
+  //add dropdown menu and icon element
+  const navListWithChildren = footer.querySelector('.nav-lists-with-child');
+  [...navListWithChildren.children].forEach(child => {
+    const accordionArrow = document.createElement('i');
+    accordionArrow.className = 'arrow';
+    child.insertBefore(accordionArrow, child.firstChild);  
+  });
+
+  //add show hide event for mobile and tablet
+  if(window.innerWidth < 1280){
```
Comment (mkbansal1) [inline blocks/footer/footer.js]: Add a method (isDesktop()) in utils.js to check if current mode is desktop mode or not.

File: blocks/footer/footer.css [hunk 3]
```diff
@@ -1,26 +1,178 @@
...
+  border-bottom: 1px solid var(--border-color);
+  display: inline-block;
+  padding: 1.063rem 0;
+  color: var(--body-text-color);
+  font-size: var( --body-font-size-xs);
+  font-weight: bold;
+}
+
+footer .footer .footer-nav .nav-lists.nav-lists-with-child > li:first-child {
+  border-top: 1px solid rgba(255, 255, 255, 0.5);
```
Comment (mkbansal1) [inline blocks/footer/footer.css]: use hax code instead of rgba

File: blocks/footer/footer.css [hunk 4]
```diff
@@ -1,26 +1,178 @@
...
+}
+
+footer .footer .footer-nav .nav-lists > li a {
+  text-decoration: none;
+  color: var(--body-text-color);
+}
+
+footer .footer .footer-nav .nav-lists > li .arrow {
+  border: solid white;
+  border-width: 0px 2px 2px 0;
```
Comment (mkbansal1) [inline blocks/footer/footer.css]: no need to add "px" before 0.

File: blocks/footer/footer.css [hunk 5]
```diff
@@ -1,26 +1,178 @@
...
+  padding: 0.25rem;
+  right: 10px;
+  position: absolute;
+  transform: rotate(45deg);
+  -webkit-transform: rotate(45deg);
+  transition: 0.4s all ease-in-out;
+}
+
+footer .footer .footer-nav .nav-lists > li ul {
+  list-style: none;
```
Comment (mkbansal1) [inline blocks/footer/footer.css]: list-style is already set as none at line 9, no need to repeat same rule again.

File: blocks/footer/footer.css [hunk 6]
```diff
@@ -1,26 +1,178 @@
...
+  font-size: var( --body-font-size-xs);
+  font-weight: bold;
+}
+
+footer .footer .footer-nav .nav-lists.nav-lists-with-child > li:first-child {
+  border-top: 1px solid rgba(255, 255, 255, 0.5);
+}
+
+footer .footer .footer-nav .nav-lists > li a {
+  text-decoration: none;
```
Comment (mkbansal1) [inline blocks/footer/footer.css]: no need to set text-decoration here, its coming from base theme.

**Review by mkbansal1 — COMMENTED** [review]
- Mobile mode - add hand icon on hover of accordion 
- Links hover should be underline
- Fix lint issues, Many issues with lint.
image

File: blocks/footer/footer.css (hunk 6 above)
Comment (akki761) [reply blocks/footer/footer.css]: done

**Comment by akki761** [comment]
@mkbansal1 for mobile mode, we do not really need cursor since it is a touch device.

File: styles/styles.css (hunk 1 above)
Comment (mkbansal1) [reply styles/styles.css]: this is corrected

File: scripts/utils.js [hunk 7]
```diff
@@ -62,3 +62,11 @@ export function decorateLinkedPictures(container) {
       a.setAttribute('title', txt);
     });
 }
+
+/**
+ * Checks whether device is desktop or not.
+ * 
+ */
+export function isDesktop() {
+  return window.innerWidth < 1280;
```
Comment (mkbansal1) [inline scripts/utils.js]: this method should return true if screen size is desktop screen, but currently its returning false.

**Comment by mkbansal1** [comment]
> @mkbansal1 for mobile mode, we do not really need cursor since it is a touch device.

ok

**Review by mkbansal1 — APPROVED** [review]

**Pull request merged**